public/app/features/dashboard/api/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { config } from '@grafana/runtime';
+import { DashboardV2Spec } from '@grafana/schema/dist/esm/schema/dashboard/v2alpha0/dashboard.gen';
+import { DashboardDTO } from 'app/types';
+
+import { DashboardWithAccessInfo } from './types';
+
+export function getDashboardsApiVersion() {
+  console.log(
+    'Dashboard API version:',
+    config.featureToggles.dashboardScene,
+    config.featureToggles.kubernetesDashboards,
+    config.featureToggles.useV2DashboardsAPI
+  );
+  // if dashboard scene is disabled, use legacy API response for the old architecture
+  if (!config.featureToggles.dashboardScene) {
+    // for old architecture, use v0 API for k8s dashboards
+    if (config.featureToggles.kubernetesDashboards) {
+      return 'v0';
+    }
+
+    return 'legacy';
+  }
```

Comment:
This builds on an assumption that legacy dashboard architecture will not be updated to hadle v2 schema. This means, that with `dashboardScene` toggle disabled, the API will fall back to the one that supports the current schema.